### PR TITLE
fix(Table): Table header truncation

### LINKED ISSUE #473: Last row in table when expanded then need to be scrolled
Now :
image

Should be like this when last row expanded

image

### LINKED ISSUE #500: Table: fix column headers in smaller screens

### PR DESCRIPTION
**Pull request by FlyersPh9**
- Table header truncates rather than wrapping to a new line. Must be placed within `<div class="iui-cell-label">`. Closes #500.

### CODE REVIEW

**Comment by mayank99** [comment]
Can you split this into two PRs? I don't think the scroll snapping will be compatible with react-table and our animations, so it would be better to handle and review it separately.

https://github.com/iTwin/iTwinUI/blob/c30031b9d52f0be93a64712dfdfb45be5aa64424/CONTRIBUTING.md?plain=1#L129

File: src/table/table.scss [hunk 1]
```diff
@@ -35,54 +35,62 @@
...
-  .iui-cell:not(.iui-slot) {
-    @include iui-focus;
-
-    &.iui-actionable {
-      cursor: pointer;
+  .iui-cell {
+    &-label {
+      white-space: nowrap;
+      overflow: hidden;
+      text-overflow: ellipsis;
```
Comment (mayank99) [inline src/table/table.scss:42]: I don't think this is a proper fix. 
image

What was wrong with `flex-wrap`? I know it wouldn't automagically handle the sort icon, but that was part of the task.

Comment (FlyersPh9) [reply src/table/table.scss:42]: I first tried applying `white-space: nowrap; flex-wrap: wrap;` but wasn't very happy with how it was looking.
Screen Shot 2022-05-02 at 12 51 59 PM

**Comment by FlyersPh9** [comment]
> Can you split this into two PRs? I don't think the scroll snapping will be compatible with react-table and our animations, so it would be better to handle and review it separately.
> 
> https://github.com/iTwin/iTwinUI/blob/c30031b9d52f0be93a64712dfdfb45be5aa64424/CONTRIBUTING.md?plain=1#L129

Removed the scroll snapping portion.

File: src/table/table.scss (hunk 1 above)
Comment (bentleyvk) [reply src/table/table.scss:42]: I agree with Mayank, truncated headers doesn't give any info to the user.

**Pull request closed without merging**